src/Tools/dotnet-monitor/CapabilitiesPostConfigureOptions.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Diagnostics.Monitoring.Options;
+using Microsoft.Diagnostics.Monitoring.WebApi;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Diagnostics.Monitoring
+{
+    public class CapabilityPostConfigureOptions :
```

Comment:
Consider pushing this to base class functionality without any vector based storage. As it is right now, this class is instantiated 4 times, each instance supporting all 4 choices, but only actually being called against one.

Reply:
I refactored this out, let me know if you had something else in mind.